### PR TITLE
Implement Multiple Session Handler

### PR DESCRIPTION
**Pull request by gayashanbc**
Related https://github.com/wso2/product-is/issues/6012

### CODE REVIEW

File: components/org.wso2.carbon.identity.application.authentication.handler.session/src/main/java/org/wso2/carbon/identity/application/authentication/handler/session/internal/ActiveSessionsLimitHandlerServiceComponent.java [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+        } catch (Throwable e) {
+            log.error("ActiveSessionsLimitHandler Authenticator bundle activation Failed", e);
+        }
+    }
+
+    @Deactivate
+    protected void deactivate(ComponentContext ctxt) {
+
+        if (log.isDebugEnabled()) {
+            log.info("ActiveSessionsLimitHandler bundle is deactivated");
```
[the remark this answers is not in the document]
Comment (gayashanbc) [reply components/org.wso2.carbon.identity.application.authentication.handler.session/src/main/java/org/wso2/carbon/identity/application/authentication/handler/session/internal/ActiveSessionsLimitHandlerServiceComponent.java]: Fixed in https://github.com/wso2-extensions/identity-local-auth-basicauth/pull/61/commits/96500f3d5a2a2ea4439083c131febf9e6c3b353f

File: components/org.wso2.carbon.identity.application.authentication.handler.session/src/main/java/org/wso2/carbon/identity/application/authentication/handler/session/internal/ActiveSessionsLimitHandlerServiceComponent.java [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+    @Reference(
+            name = "realm.service",
+            service = RealmService.class,
+            cardinality = ReferenceCardinality.MANDATORY,
+            policy = ReferencePolicy.DYNAMIC,
+            unbind = "unsetRealmService"
+    )
+    protected void setRealmService(RealmService realmService) {
+
+        log.debug("Setting the Realm Service");
```
[the remark this answers is not in the document]
Comment (gayashanbc) [reply components/org.wso2.carbon.identity.application.authentication.handler.session/src/main/java/org/wso2/carbon/identity/application/authentication/handler/session/internal/ActiveSessionsLimitHandlerServiceComponent.java]: Fixed in https://github.com/wso2-extensions/identity-local-auth-basicauth/pull/61/commits/96500f3d5a2a2ea4439083c131febf9e6c3b353f

File: pom.xml [hunk 3]
```diff
@@ -379,7 +380,7 @@
         <carbon.base.imp.pkg.version.range>[1.0.0, 2.0.0)</carbon.base.imp.pkg.version.range>
 
         <!-- Carbon Identity Framework version -->
-        <carbon.identity.framework.version>5.14.0</carbon.identity.framework.version>
+        <carbon.identity.framework.version>5.14.57</carbon.identity.framework.version>
         <carbon.identity.framework.imp.pkg.version.range>[5.14.0, 6.0.0)
```
[the remark this answers is not in the document]
Comment (gayashanbc) [reply pom.xml]: Fixed in https://github.com/wso2-extensions/identity-local-auth-basicauth/pull/61/commits/96500f3d5a2a2ea4439083c131febf9e6c3b353f

File: components/org.wso2.carbon.identity.application.authentication.handler.session/src/main/java/org/wso2/carbon/identity/application/authentication/handler/session/internal/ActiveSessionsLimitHandlerServiceComponent.java [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+public class ActiveSessionsLimitHandlerServiceComponent {
+
+    private static final Log log = LogFactory.getLog(ActiveSessionsLimitHandlerServiceComponent.class);
+
+    @Activate
+    protected void activate(ComponentContext ctxt) {
+
+        try {
+            ActiveSessionsLimitHandler activeSessionsLimitHandler = new ActiveSessionsLimitHandler();
+            ctxt.getBundleContext().registerService(ApplicationAuthenticator.class.getName(), activeSessionsLimitHandler, null);
```
Comment (JKAUSHALYA) [inline components/org.wso2.carbon.identity.application.authentication.handler.session/src/main/java/org/wso2/carbon/identity/application/authentication/handler/session/internal/ActiveSessionsLimitHandlerServiceComponent.java]: Should be exceeding 120 lines count.

File: components/org.wso2.carbon.identity.application.authentication.handler.session/src/main/java/org/wso2/carbon/identity/application/authentication/handler/session/internal/ActiveSessionsLimitHandlerServiceComponent.java [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+            cardinality = ReferenceCardinality.MANDATORY,
+            policy = ReferencePolicy.DYNAMIC,
+            unbind = "unsetUserSessionManagementService"
+    )
+    protected void setUserSessionManagementService(UserSessionManagementService userSessionManagementService) {
+
+        if (log.isDebugEnabled()) {
+            log.debug("UserSessionManagementService is set in the conditional authentication user functions bundle");
+        }
+        ActiveSessionsLimitHandlerServiceHolder.getInstance().setUserSessionManagementService(userSessionManagementService);
```
Comment (JKAUSHALYA) [inline components/org.wso2.carbon.identity.application.authentication.handler.session/src/main/java/org/wso2/carbon/identity/application/authentication/handler/session/internal/ActiveSessionsLimitHandlerServiceComponent.java]: Should be exceeding 120 lines count.

File: components/org.wso2.carbon.identity.application.authentication.handler.session/src/main/java/org/wso2/carbon/identity/application/authentication/handler/session/internal/ActiveSessionsLimitHandlerServiceComponent.java (hunk 4 above)
Comment (gayashanbc) [reply components/org.wso2.carbon.identity.application.authentication.handler.session/src/main/java/org/wso2/carbon/identity/application/authentication/handler/session/internal/ActiveSessionsLimitHandlerServiceComponent.java]: Resolved in https://github.com/wso2-extensions/identity-local-auth-basicauth/pull/68/commits/aa9404e4bdae50bedd69f6a1492773da1a76bc80

File: components/org.wso2.carbon.identity.application.authentication.handler.session/src/main/java/org/wso2/carbon/identity/application/authentication/handler/session/internal/ActiveSessionsLimitHandlerServiceComponent.java (hunk 5 above)
Comment (gayashanbc) [reply components/org.wso2.carbon.identity.application.authentication.handler.session/src/main/java/org/wso2/carbon/identity/application/authentication/handler/session/internal/ActiveSessionsLimitHandlerServiceComponent.java]: Resolved in https://github.com/wso2-extensions/identity-local-auth-basicauth/pull/68/commits/aa9404e4bdae50bedd69f6a1492773da1a76bc80

**Pull request merged**